acceptance/framework/helpers/helpers.go

```diff
@@ -84,11 +86,10 @@ func SetupInterruptHandler(cleanup func()) {
 	}()
 }
 
-// Cleanup will both register a cleanup function with t
-// and SetupInterruptHandler to make sure resources get cleaned up
-// if an interrupt signal is caught.
-func Cleanup(t *testing.T, noCleanupOnFailure bool, noCleanup bool, cleanup func()) {
-	t.Helper()
+// CleanupWithOnFailure will both register a cleanup function with t and SetupInterruptHandler to make sure resources
+// get cleaned up if an interrupt signal is caught.
+func CleanupWithOnFailure(t terratesting.TestingT, noCleanupOnFailure bool, noCleanup bool, cleanup func()) {
```

Comment:
Note that if `t` is a `retry.R` then `Failed` will only return true if the last retry attempt failed. This means that you probably shouldn't add a cleanup func like this in a retry block **unless you use the `retry.WithImmediateCleanup` method to run cleanup funcs between each retry attempt.